src/box/lua/schema.lua

```diff
@@ -601,11 +601,20 @@ box.schema.index = {}
 
 local function update_index_parts_1_6_0(parts)
     local result = {}
-    if #parts % 2 ~= 0 then
+    if table.getn(parts) % 2 ~= 0 then
```

Comment:
I've googled and it turns out that getn() function is sort of deprecated: http://www.lua.org/manual/5.1/manual.html#7.2
I've also grepped almost no usages of getn in our codebase. I guess @igormunkin or @Gerold103 may provide more details on this subj. But I suggest using '#' instead.

Reply:
@Korablev77, considering the Lua reference manual section you've mentioned and [LuaJIT implementation](https://github.com/tarantool/luajit/blob/tarantool/src/lib_table.c#L50L55) `table.getn(t)` yields `#t`. Totally agree with your suggestion.

Reply:
Oops, initially I found that '#' is undefined when the table is not a sequence, but lua reference says it is defined now. So yes, my mistake :) Fixed

Reply:
@mary3000, `#` is still undefined when the table is not a sequence. However, there is no need to use `table.getn` instead of `#`, since both interfaces return the same value -- this was the main point.

Reply:
@igormunkin hmm, [3.4.7 paragraph](https://www.lua.org/manual/5.4/manual.html#3.4.7) describes definition for a general table, although the result is non-deterministic. Or I need to look at another lua version?

We expect even-sized sequence in this function, and I think this check is ok for our purposes: if the table is not a sequence at all, either there will be error now, or in the next loop (which now doesn't use '#', but rather iterates over all pairs). If the table is an odd-sized sequence, then we correctly throw an error here.

Reply:
> @igormunkin hmm, [3.4.7 paragraph](https://www.lua.org/manual/5.4/manual.html#3.4.7) describes definition for a general table, although the result is non-deterministic. Or I need to look at another lua version?

@mary3000, you need [this one](http://www.lua.org/manual/5.1/manual.html#2.5.5) since LuaJIT implements* Lua 5.1, not Lua 5.4. Anyway, IIRC, there are no changes between these versions related to the table length definition, but yours one looks much more verbose and clear. When I said "undefined" I meant exactly non-deterministic, so you're right the result is _implementation-defined_. There are nice examples in almost yearling [ticket](https://github.com/LuaJIT/LuaJIT/issues/527):
```bash
$ lua5.1
Lua 5.1.5  Copyright (C) 1994-2012 Lua.org, PUC-Rio
> function a(...)
>> local x = {...}
>> print(#x)
>> end
> a(1, 2, nil, 4, 5)
5
> 
$ luajit
LuaJIT 2.0.5 -- Copyright (C) 2005-2017 Mike Pall. http://luajit.org/
JIT: ON CMOV SSE2 SSE3 SSE4.1 fold cse dce fwd dse narrow loop abc sink fuse
> function a(...)
>> local x = {...}
>> print(#x)
>> end
> a(1, 2, nil, 4, 5)
2
```

> We expect even-sized sequence in this function, and I think this check is ok for our purposes: if the table is not a sequence at all, either there will be error now, or in the next loop (which now doesn't use '#', but rather iterates over all pairs). If the table is an odd-sized sequence, then we correctly throw an error here.

Neat.

\* Strictly saying, it's not, but Mike tried his best.